src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -858,30 +858,30 @@ void GcsActorManager::OnActorCreationSuccess(const std::shared_ptr<GcsActor> &ac
   }
   actor->UpdateState(rpc::ActorTableData::ALIVE);
   auto actor_table_data = actor->GetActorTableData();
+  // Invoke all callbacks for all registration requests of this actor (duplicated
+  // requests are included) and remove all of them from
+  // actor_to_create_callbacks_.
+  auto iter = actor_to_create_callbacks_.find(actor_id);
+  if (iter != actor_to_create_callbacks_.end()) {
+    for (auto &callback : iter->second) {
+      callback(actor);
+    }
+    actor_to_create_callbacks_.erase(iter);
+  }
+
+  // We should register the entry to the in-memory index before flushing them to
+  // GCS because otherwise, there could be timing problems due to asynchronous Put.
+  auto worker_id = actor->GetWorkerID();
+  auto node_id = actor->GetNodeID();
+  RAY_CHECK(!worker_id.IsNil());
+  RAY_CHECK(!node_id.IsNil());
+  RAY_CHECK(created_actors_[node_id].emplace(worker_id, actor_id).second);
```

Comment:
The issue here is that if actor is destroyed at this timing, actors won't be in both `created_map` and `creating_map` as well as `leasing_map` and `pending_map`. That will break the destroy actor method.